### ISSUE TITLE
[Bug] Diagnostics not triggered on file change

### ISSUE THREAD

**Issue by JorgeHerreraU** [seblj/roslyn.nvim#76]
When you change a reference behavior between two files, diagnostics are not triggered until you start editing the file.

https://github.com/user-attachments/assets/94a05f54-0ed1-4318-b3d7-a65a085d113b

Current configuration

```lua
    -- csharp
    require('roslyn').setup {
      config = {
        capabilities = capabilities,
        on_attach = function(client, bufnr)
          vim.bo[bufnr].tabstop = 4
          vim.bo[bufnr].shiftwidth = 4
          vim.bo[bufnr].expandtab = true
          vim.bo[bufnr].softtabstop = 4
          vim.api.nvim_create_autocmd('BufEnter', {
            buffer = bufnr,
            callback = function()
              vim.lsp.codelens.refresh { bufnr = 0 }
              -- workaround for diagnostics not being triggered
              -- client.request('textDocument/diagnostic', {
              --   textDocument = vim.lsp.util.make_text_document_params(),
              -- }, nil, bufnr)
            end,
          })
        end,
        filewatching = true,
      },
    }
```

**Comment by seblj** [comment]
https://github.com/seblj/roslyn.nvim/wiki#diagnostic-refresh

**Comment by seblj** [comment]
Diagnostics are supported with a hack since the server is implementing something that neovim doesn't fully support yet and vice versa
